### DISCUSSION TITLE
server_name is not being used as the Server header

### DISCUSSION THREAD

**Discussion by Siecje** [cherrypy/cheroot#45] [Q&A] [answered]
* **I'm submitting a ...**
[x] bug report
[ ] feature request
[ ] question about the decisions made in the repository

* **Do you want to request a *feature* or report a *bug*?**
Report a bug

* **What is the current behavior?**

The server_name keyword argument is not used as the `Server` header value  in the response.

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a screenshots and logs of the problem. If you can, show us your code.**

In one terminal run
```
from cheroot.wsgi import Server
import cherrypy

class HelloWorld(object):
    @cherrypy.expose
    def index(self):
        return "Hello World!"


cherrypy.tree.mount(HelloWorld(), "/")
server = Server(('localhost', 8090), cherrypy.tree, server_name = 'Siecje')
server.start()
```

In another terminal run
```
import requests
r = requests.get('http://localhost:8090')
print(r.headers['Server])
# I believe it should be 'Siecje'
```

* **What is the expected behavior?**

The server_name should be used as the `Server` header value.

* **What is the motivation / use case for changing the behavior?**

In the docs for `wsgi.Server` it says
`server_name (str): web server name to be advertised via Server HTTP header`

* **Please tell us about your environment:**

- Python version: 2.7.13
- Cheroot version: 5.8.3
- CherryPy version: 11.0.0
- OS: Linux Ubuntu 16.04 64bit
- Browser: [all]

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, e.g. stackoverflow, gitter, etc.)

**Comment by webknjaz** [comment]
What do you get for `Server` header instead?
I guess it might be overwritten within cherrypy: [_cpwsgi_server.py:65](https://github.com/cherrypy/cherrypy/blob/c87bc9f73740dfe35e4d2ca9082dccb6278bdc53/cherrypy/_cpwsgi_server.py#L65)

**Comment by Siecje** [comment]
The value I get for `r.headers['Server']` is `'CherryPy/11.0.0'`

**Comment by webknjaz** [comment]
Which means, I'm right. Have you tried setting it using CherryPy config?

**Comment by Siecje** [comment]
How should I set it?

When I stepped through it was being set here https://github.com/cherrypy/cherrypy/blob/master/cherrypy/_cprequest.py#L897

And not overwritten. 

In `CherryPy` it is available here as `self.environ['SERVER_NAME']`.

https://github.com/cherrypy/cherrypy/blob/master/cherrypy/_cpwsgi.py#L310

**Comment by webknjaz** [comment]
When using CherryPy, you should use [`CPWSGIServer`](https://github.com/cherrypy/cherrypy/blob/master/cherrypy/_cpwsgi_server.py#L31), which sets up everything correctly. And there's also mechanism in CherryPy for setting up `server_name` via config.

Why do you use underlying layer (which is Cheroot) after all?

**Comment by Siecje** [comment]
I was using
```Python
from cherrypy.wsgiserver import CherryPyWSGIServer, HTTPServer
HTTPServer.version = "Siecje"
```
before the `cheroot` split.


So I should replace that with from `cherrypy._cpwsgi_server import CPWSGIServer` ?

**Comment by Siecje** [comment]
Where do you specify `server` header value?

**Comment by webknjaz** [comment, marked as answer]
Well, this depends on whether you need CherryPy framework or you just need Cheroot WSGI/HTTP layer. If you can/want to drop CherryPy dependency, it should be or to use `cheroot.wsgi.Server` (or `cheroot.server.HTTPServer`) and pass `server_name` there directly.

If you use CherryPy, I'm not sure what you are trying to achieve. Can you show me your code example or at least explain your intentions when using custom WSGI Server?

If using normal CherryPy setup, it should be possible to [set `response.headers.server` config option like this](https://stackoverflow.com/questions/135317/cherrypy-server-name-tag/15035521#15035521).

**Comment by Siecje** [comment]
Thanks setting 
`cherrypy.config.update({'response.headers.server': 'Siecje'})`
worked.
